crates/services/gas_price_service/src/v1/service.rs

```diff
@@ -304,11 +314,60 @@ where
         let new_algo = self.algorithm_updater.algorithm();
         tracing::debug!("Updating gas price: {}", &new_algo.calculate());
         self.shared_algo.update(new_algo);
+        let best_recorded_height = new_recorded_height.or(old_recorded_height);
+        Self::record_metrics(&metadata, gas_price, best_recorded_height);
         // Clear the buffer after committing changes
         self.da_block_costs_buffer.clear();
         Ok(())
     }
 
+    fn record_metrics(
+        metadata: &UpdaterMetadata,
+        gas_price: u64,
+        recorded_height: Option<BlockHeight>,
+    ) {
+        if let UpdaterMetadata::V1(v1_metadata) = metadata {
+            let metrics = gas_price_metrics();
+            let real_gas_price_i64 = gas_price.try_into().unwrap_or(i64::MAX);
+            let exec_gas_price_i64 = v1_metadata
+                .new_exec_gas_price()
+                .try_into()
+                .unwrap_or(i64::MAX);
```

Comment:
Nit: I generally prefer explicit conversions for readability. I.e. `i64::try_from()`.

I'm also uncertain if it makes sense to report `i64::MAX` if we're out of range. Wouldn't it be better to not report any metrics and log an error in this case?